examples/cfd/01_convection_revisited.ipynb

```diff
@@ -391,10 +391,78 @@
+      "#define _POSIX_C_SOURCE 200809L\n",
+      "#define MIN(a,b) (((a) < (b)) ? (a) : (b))\n",
+      "#define MAX(a,b) (((a) > (b)) ? (a) : (b))\n",
+      "#define START_TIMER(S) struct timeval start_ ## S , end_ ## S ; gettimeofday(&start_ ## S , NULL);\n",
+      "#define STOP_TIMER(S,T) gettimeofday(&end_ ## S, NULL); T->S += (double)(end_ ## S .tv_sec-start_ ## S.tv_sec)+(double)(end_ ## S .tv_usec-start_ ## S .tv_usec)/1000000;\n",
+      "\n",
+      "#include \"stdlib.h\"\n",
+      "#include \"math.h\"\n",
+      "#include \"sys/time.h\"\n",
+      "\n",
+      "struct dataobj\n",
+      "{\n",
+      "  void *restrict data;\n",
+      "  int * size;\n",
+      "  int * npsize;\n",
+      "  int * dsize;\n",
+      "  int * hsize;\n",
+      "  int * hofs;\n",
+      "  int * oofs;\n",
+      "} ;\n",
+      "\n",
+      "struct profiler\n",
+      "{\n",
+      "  double section0;\n",
+      "  double section1;\n",
+      "  double section2;\n",
+      "} ;\n",
+      "\n",
+      "\n",
+      "int Kernel(const float dt, const float h_x, const float h_y, struct dataobj *restrict u_vec, const int i0x_ltkn, const int i0x_rtkn, const int i0y_ltkn, const int i0y_rtkn, const int time_M, const int time_m, const int x_M, const int x_m, const int y_M, const int y_m, struct profiler * timers)\n",
+      "{\n",
+      "  float (*restrict u)[u_vec->size[1]][u_vec->size[2]] __attribute__ ((aligned (64))) = (float (*)[u_vec->size[1]][u_vec->size[2]]) u_vec->data;\n",
+      "\n",
+      "  for (int time = time_m, t0 = (time)%(2), t1 = (time + 1)%(2); time <= time_M; time += 1, t0 = (time)%(2), t1 = (time + 1)%(2))\n",
+      "  {\n",
+      "    /* Begin section0 */\n",
+      "    START_TIMER(section0)\n",
+      "    for (int i0x = i0x_ltkn + x_m; i0x <= -i0x_rtkn + x_M; i0x += 1)\n",
```

Comment:
as I had pointed out, we shouldn't be regenerating the entire file, but rather only the portions that actually changed. It's tedious, I know, but it makes sure there are no nasty regressions